src/index-utils.ts

```diff
@@ -338,33 +242,24 @@ function getMatchingLogics(actionType: ActionType, modifiedFields: DocumentData,
   });
 }
 
-async function runLogic(action: Action, matchingLogics: LogicConfig[], nextPageMarkers: (object | undefined)[],
-  sharedMap: Map<string, any>, logicResults: LogicResult[]) {
-  for (let i = matchingLogics.length - 1; i >= 0; i--) {
+async function runLogic(txnGet: TxnGet, action: Action, matchingLogics: LogicConfig[], sharedMap: Map<string, any>,
```

Comment:
I think this function is redundant with runBusinessLogics.  Let's transfer this code block to inside runBusinessLogics or put the for loop inside runBusinessLogics and just pass one logic here so that runLogic name will make more sense